README.md

```diff
@@ -223,6 +225,36 @@ app.use(function (err, req, res, next) {
 })
 ```
 
+### Verifying a token
+
+If you need to pass the token in a query string rather than an
+HTTP header, you can use the `req.verifyToken(token)` function
+provided by this middleware. This is useful when you want to
+use this library with an oauth callback.
+
+```js
+var csrf = require('csurf')
+var express = require('express')
+
+var app = express()
+app.use(csrf())
+
+app.get('/oauth_callback', function(req, res) {
+  try {
+    req.verifyToken(req.query.state); // throws
```

Comment:
The value option (https://github.com/expressjs/csurf/blob/master/README.md#value) can take a function to pull the token from anywhere you want, and verify it automatically. This includes from req.query.state